tests/test_group.rs

```diff
@@ -159,109 +212,156 @@ fn basic_group_setup() {
+    // Generate KeyPackages
+    let alice_key_package_bundle = KeyPackageBundle::new(
+        ciphersuite_name,
+        alice_signature_key,
+        Credential::from(MLSCredentialType::Basic(alice_credential)),
+        mandatory_extensions.clone(),
+    );
 
-    let mut group_alice = Group::new(alice_identity, GroupId::random(), config);
+    let bob_key_package_bundle = KeyPackageBundle::new(
+        ciphersuite_name,
+        bob_signature_key,
+        Credential::from(MLSCredentialType::Basic(bob_credential)),
+        mandatory_extensions,
+    );
+    let bob_key_package = bob_key_package_bundle.get_key_package();
 
-    // Alice sends a message to herself
-    let message_alice = [1, 2, 3];
-    let mls_plaintext = group_alice.create_application_message(&message_alice, Some(&[4, 5, 6]));
-    let encrypted_message = group_alice.encrypt(&mls_plaintext);
-    let decrypted_mls_plaintext = group_alice.decrypt(&encrypted_message);
-    assert_eq!(mls_plaintext, decrypted_mls_plaintext);
+    // Alice creates a group
+    let group_id = [1, 2, 3, 4];
+    let mut group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
 
     // Alice adds Bob
-    let _bob_add_proposal = group_alice.create_add_proposal(&bob_key_package, None);
-
-    let (commit1, ms1, welcome_bundle_alice_bob) = group_alice.create_commit(None);
+    let bob_add_proposal = group_alice_1234.create_add_proposal(
+        group_aad,
+        alice_signature_key,
+        bob_key_package.clone(),
+    );
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(
+            group_aad,
+            alice_signature_key,
+            epoch_proposals.clone(),
+            false, // TODO force update seems broken
+        ) {
+        Ok(c) => c,
+        Err(e) => panic!("Error creating commit: {:?}", e),
+    };
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
+        _ => panic!(),
+    };
+    assert!(commit.path.is_none());
 
-    group_alice.process_commit(commit1);
+    match group_alice_1234.apply_commit(mls_plaintext_commit, epoch_proposals, vec![]) {
+        Ok(_) => {}
+        Err(e) => panic!("Error applying commit: {:?}", e),
+    };
+    let ratchet_tree = group_alice_1234.tree().get_public_key_tree();
 
-    let mut group_bob = Group::new_from_welcome(
-        bob_identity,
-        welcome_bundle_alice_bob.unwrap(),
+    let mut group_bob = match MlsGroup::new_from_welcome(
+        welcome_bundle_alice_bob_option.unwrap(),
+        Some(ratchet_tree),
         bob_key_package_bundle,
-    );
+    ) {
+        Ok(group) => group,
+        Err(e) => panic!("Error creating group from Welcome: {:?}", e),
+    };
 
-    assert_eq!(group_alice.tree.nodes, group_bob.tree.nodes);
-    group_alice.tree.print(&format!("\n{:?}", ms1));
+    assert_eq!(
```

Comment:
This looks like a pretty crucial test. It should probably get an explanation.
Also, the `_print_tree` should only be interesting if this fails. But it won't be reached in the case the assert fails.